src/Plugin/rate_limit/RateLimit.php

```diff
@@ -38,7 +39,12 @@
    */
   public function __construct(array $configuration, $plugin_id, $plugin_definition) {
     parent::__construct($configuration, $plugin_id, $plugin_definition);
-    $this->period = $configuration['period'];
+    try {
+      $this->period = new \DateInterval($configuration['period']);
+    }
+    catch (\Exception $e) {
+      throw new ServerConfigurationException(sprintf('Invalid rate limit period: %s.', $configuration['period']));
```

Comment:
Maybe: ``Invalid rate limit period: %s. Should be a valid format of \DateInterval.``